code/parachain/runtime/common/src/xcmp.rs

```diff
@@ -383,3 +340,35 @@ impl<X, Y, Treasury: TakeRevenue, Z> Drop for TransactionFeePoolTrader<X, Y, Tre
 		}
 	}
 }
+pub struct RelayReserveFromParachain;
+impl FilterAssetLocation for RelayReserveFromParachain {
+	fn filter_asset_location(asset: &MultiAsset, origin: &MultiLocation) -> bool {
+		AbsoluteReserveProvider::reserve(asset) == Some(MultiLocation::parent()) &&
+			matches!(origin, MultiLocation { parents: 1, interior: X1(Parachain(_)) })
+	}
+}
+
+/// Estimates outgoing fees on target chain in transferred token
+pub struct OutgoingFee<Registry: RemoteAssetRegistryInspect> {
+	_marker: PhantomData<Registry>,
+}
+
+impl<
+		Registry: RemoteAssetRegistryInspect<
+			AssetId = CurrencyId,
+			AssetNativeLocation = XcmAssetLocation,
+			Balance = Balance,
+		>,
+	> OutgoingFee<Registry>
+{
+	pub fn outgoing_fee(location: &MultiLocation) -> Option<Balance> {
+		match (location.parents, location.first_interior()) {
+			(1, None) => Some(400_000_000_000),
+			(1, Some(Parachain(id))) => {
+				let location = XcmAssetLocation::new(location.clone());
+				Registry::min_xcm_fee(ParaId::from(*id), location).or(Some(u128::MAX))
```

Comment:
Will recheck this part and tell what Acala does with this. That was some stuff taken from Acala some time ago.

Reply:
Max prevented sending assets away. will set it to resonable default

Reply:
If the registry has no min fee defined it might make sense to prevent the assets from being sent anyway. We just need to prevent a scenario of assets getting trapped I guess

Reply:
I will add test which proves that assets just failed to sent, not trapped.

Reply:
As as I see Acala does not prevents sending. So set to None as they do. But improved us for a case.